docs/reference/remap.cue

```diff
@@ -52,17 +51,589 @@ remap: {
+				expression sets `user.first_name` to `"Feldman"` *if* that field exists; if not, the
+				`user.last_name` field is set to `"Feldman"` instead.
+
+				```
+				.user.(first_name | last_name) = "Feldman"
+				```
+
+				### Quoted paths
+
+				In the examples above, all paths have used literals à la `.foo.bar`. But path
+				segments can also be quoted, as in this example:
+
+				```
+				user.preferences."favorite color" = "chartreuse"
+				```
+
+				Quoted paths are particularly useful when keys need to contain whitespace.
+
+				### Indexing
+
+				Values inside TRL arrays can be accessed via index (beginning with 0). For the array
+				`$primary = ["magenta", "yellow", "cyan"]`, `$primary[0]` would yield `"magenta"`.
+
+				You can also assign values to arrays by index:
+
+				```
+				$stooges = ["Larry", "Moe"]
+				$stooges[2] = "Curly"
+				```
+
+				You can even assign values to arbitrary indices in arrays; any indices that need to
+				be created are back-filled as `null`. For example, if the `hobbies` field doesn't
+				exist, the expression `.hobbies[2] = "Pogs"` sets `hobbies` to `[null, null,
+				"Pogs"]`.
+
+				### Combined
+
+				All of the path methods above can be combined in any way. Here's an example of a
+				complex path:
+
+				```
+				.transaction.(metadata | info).orders[0] = "a1b2c3d4e5f6"
+				```
+
+				This sets the first element of `.transaction.metadata.orders` to `"a1b2c3d4e5f6"`
+				*if* `.transaction.metadata` exist; if not, it sets `.transaction.info.orders` to
+				that value.
+				"""
+			examples: [
+				".",
+				".status_code",
+				#".message.event."time of occurrence""#,
+				".transaction.id",
+				".user.hobbies[0].description",
+				".event.(time | timestamp).format",
+				#""#,
+			]
+		}
+
+		"Expressions": {
+			href: "expressions"
+
+			description: """
+				*All* expressions in TRL resolve to a value. Expressions come in four kinds, listed
+				below, each of which resolves in a different way:
+
+				Expression type | Resolves to
+				:---------------|:-----------
+				Assignment | The assigned value
+				Control flow statements | The value returned by the chosen expression
+				Boolean expressions | The returned Boolean value
+				Blocks | The value returned by the last expression in the block
+				"""
+
+			in_n_out: [
+				{in: #".request_id = "a1b2c3d4e5f6""#, out:                      "string"},
+				{in: #"if (starts_with("v1", .version)) { .version = 1 }"#, out: "integer"},
+				{in: #"contains("emergency", "this is an emergency")"#, out:     "boolean"},
+				{
+					in: """
+						$is_success = { $code = .status_code; del(.status_code); $code == 200 }
+						"""
+					out: "boolean"
+				},
+			]
+		}
+
+		"Lines": {
+			href: "lines"
+
+			description: #"""
+				TRL expressions can be split across multiple lines using a backslash (`\`):
+
+				```
+				del(.one, .two, .three, .four \
+					.five, .six)
```

Comment:
I would probably use a different example, since we're going to change the `del` function to only take a single path to delete (deleting multiple paths will be facilitated through generic traversal/iterating functionality)

Reply:
Maybe use a non-function example:

```
.foo = \
	"bar"
```